src/SprykerSdk/Zed/ComposerConstrainer/Business/Composer/ComposerJson/ComposerJsonWriter.php

```diff
@@ -42,7 +43,12 @@ public function write(array $composerJsonArray): bool
             $indentation = $this->autoDetectIndentation($composerJsonFileName);
         }
         if ($indentation !== static::INDENTATION_DEFAULT) {
-            $encodedJson = preg_replace('/^(    +?)\\1(?=[^' . str_repeat(' ', $indentation) . '])/m', '$1', $encodedJson) . "\n";
+            if (!class_exists(Printer::class)) {
```

Comment:
Please consider moving this code into separate method as this is a details of "how to write right indentation" + some precondition check. The whole ::write() method will be simplified significantly.